### PR TITLE
Raise error when loading param via response api

### LINKED ISSUE #6734: Possible to load parameters as responses
**Describe the bug**
It is possible to load any dataset through the `load_responses` and `load_parameters` end points, regardless of them being parameters or responses. This should be verified. The experiment has information about what parameters and responses are registered, and we should check that whatever we load is already registered.

**To reproduce**
Steps to reproduce the behaviour:
```python
uniform_parameter = GenKwConfig(
        name="PARAMETER",
        forward_init=False,
        template_file="",
        transfer_function_definitions=[
            "KEY1 UNIFORM 0 1",
        ],
        output_file="kw.txt"
)
experiment = storage.create_experiment(
    parameters=[uniform_parameter],
)
prior = storage.create_ensemble(
    experiment,
    ensemble_size=1,
    iteration=0,
    name="prior",
)

prior.save_parameters(
    "PARAMETER",
    0,
    xr.Dataset(
        {
            "values": ("names", [1.0]),
            "transformed_values": ("names", [1.0]),
            "names": ["KEY_1"],
        }
    ),
)
prior.load_responses("PARAMETER", (0,))  # <----- This is a problem, as this is a parameter, not a response
```

### PR DESCRIPTION
**Pull request by dafeda**
**Issue**
Resolves #6734 

## Pre review checklist

- [ ] Read through the code changes carefully after finishing work
- [ ] Make sure tests pass locally (after every commit!)
- [ ] Prepare changes in small commits for more convenient review (optional)
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Updated documentation
- [ ] Ensured that unit tests are added for all new behavior (See
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: tests/unit_tests/storage/test_local_ensemble.py [hunk 1]
```diff
@@ -66,7 +67,9 @@ def test_that_load_responses_throws_exception(tmp_path):
         experiment = storage.create_experiment()
         ensemble = storage.create_ensemble(experiment, name="foo", ensemble_size=1)
 
-        with pytest.raises(expected_exception=KeyError):
+        with pytest.raises(
+            expected_exception=ValueError, match="I_DONT_EXIST is not a response"
+        ):
```
Comment (kvashchuka) [inline tests/unit_tests/storage/test_local_ensemble.py:72]: Doesn't it now check a different thing?

Comment (dafeda) [reply tests/unit_tests/storage/test_local_ensemble.py:72]: Yes it does. The new check is more explicit I think.

Comment (kvashchuka) [reply tests/unit_tests/storage/test_local_ensemble.py:72]: Though one cannot create a response, which is a response, but doesn't exist😅 Maybe this test is redundant? It is just a suggestion, feel free to disagree

**Review by kvashchuka — APPROVED** [review]
LGTM

I had a suggestion to remove one of the tests, will leave it for you to decide

File: tests/unit_tests/storage/migration/test_block_fs_snake_oil.py [hunk 2]
```diff
@@ -74,31 +76,51 @@ def test_migrate_gen_kw(data, ensemble, parameter, ens_config):
...
+def test_migrate_summary(data, forecast, time_map, tmp_path):
     group = "/REAL_0/SUMMARY"
-    bf._migrate_summary(ensemble, forecast, time_map)
+    with open_storage(tmp_path / "storage", mode="w") as storage:
+        experiment = storage.create_experiment(
+            responses=[
+                SummaryConfig(name="summary", input_file="some_file", keys=["some_key"])
+            ]
+        )
+        ensemble = experiment.create_ensemble(name="default_0", ensemble_size=5)
```
Comment (kvashchuka) [inline tests/unit_tests/storage/migration/test_block_fs_snake_oil.py:87]: Appreciate a refactor in the tests in this file!

File: tests/unit_tests/storage/test_local_ensemble.py (hunk 1 above)
Comment (dafeda) [reply tests/unit_tests/storage/test_local_ensemble.py:72]: Not sure I follow. This makes sure `load_responses` provides a reasonable error message when users call it before responses are stored to storage, or using the wrong identifier.
Could you explain a bit more?

Comment (kvashchuka) [reply tests/unit_tests/storage/test_local_ensemble.py:72]: If I understood correctly, we are trying to load response that does not exist. And the error message now says that what we are trying to load is not a response. Is that not the case?

Comment (dafeda) [reply tests/unit_tests/storage/test_local_ensemble.py:72]: Yes, but given an instance of an `ensemble` object, we can't be sure up front whether the response we are requesting from it exists. I'm probably misunderstanding something here. Should we return a different message you think?

Comment (kvashchuka) [reply tests/unit_tests/storage/test_local_ensemble.py:72]: I retract my comment, it is good to have this test, cause it does check a different behaviour than others. Would be nice for users sake to give an explicit message on when the response does not exist, but that is outside of scope of this PR

**Pull request merged**